packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -515,17 +515,21 @@ function constructClassInstance(
   const contextType = ctor.contextType;
   if (typeof contextType === 'object' && contextType !== null) {
     if (__DEV__) {
+      const isContextConsumer =
+        contextType.$$typeof === REACT_CONTEXT_TYPE &&
+        contextType._context !== undefined;
```

Comment:
What if context value actually *is* undefined?

Reply:
Isn't the context value tracked as `_currentValue` on this `_context` element?

Reply:
Oh right.